rs/bitcoin/adapter/src/lib.rs

```diff
@@ -178,6 +181,12 @@ impl AdapterState {
     pub fn received_now(&self) {
         // Instant::now() is monotonically nondecreasing clock.
         *self.last_received_at.write() = Some(Instant::now());
+        self.awake_notify.notify_waiters();
+    }
+
+    /// Returns a clone of the notify handle.
+    pub fn awake_notifier(&self) -> Arc<Notify> {
```

Comment:
I would write this as 

```
    pub fn wait_for_request(&self) -> Notified {
        self.awake_notify.notified()
```

and in the other file I would just do

`                adapter_state.wait_for_request().await;
`